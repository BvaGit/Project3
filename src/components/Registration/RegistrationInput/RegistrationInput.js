import React, { useState} from 'react';
import Input from '../../Input';
import { validateRegField } from '/src/helpers/validation'

const RegistrationInput = ({ value, title, changeField, id, label, placeholder, error, setError }) =>{
    const [blured, setBlured] = useState(false);

    const handleBlur = () =>{
        setBlured(true);
        if(id !== 'confirm_password') setError({[id]: validateRegField(id, value) })
    }

    const handleChange = (e) => {
        changeField({ [id]: e.target.value })
        if(blured) setBlured(false)
    }

    return (
        <>
            <Input  
                labelId={label} 
                type="text" 
                name={id}
                value={value}
                title={title}
                placeholderId={placeholder} 
                minLength="1"
                maxLength="30"
                onBlur={handleBlur}
                onChange={handleChange}
            />
            {(blured && error) && <div className="inputErr">{error}</div>}
        </>
    )
}

export default RegistrationInput