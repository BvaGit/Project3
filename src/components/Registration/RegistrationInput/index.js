import RegistrationInput from './RegistrationInput';
import { connect } from 'react-redux';
import { getRegErrorByIdStore,getRegFieldByIdStore } from '../../../store/registration/selectors';
import { changeRegistrationField, changeRegistrationFieldError } from '../../../store/registration/actions';

const mapStateToProps = (state, props) => ({
    value: getRegFieldByIdStore(state, props),
    error: getRegErrorByIdStore(state, props)
})

const mapDispatchToProps = (dispatch) =>({
    changeField:(data) => dispatch(changeRegistrationField(data)),
    setError:(data) => dispatch(changeRegistrationFieldError(data)),
})

export default connect(mapStateToProps,mapDispatchToProps)(RegistrationInput)