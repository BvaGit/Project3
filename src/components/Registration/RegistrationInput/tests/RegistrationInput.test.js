import React from 'react';
import { shallow, mount } from 'enzyme';
import RegistrationInput from '../RegistrationInput';
import { IntlProvider } from "react-intl";
import { Provider } from 'react-redux';




describe('RegistrationInput', () => {
    it("Should match snapshot", () => {
        const component = shallowBiggerSmart(<RegistrationInput />)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider>core</IntlProvider>);
};