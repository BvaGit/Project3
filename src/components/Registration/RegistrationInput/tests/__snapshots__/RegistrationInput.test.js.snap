// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationInput Should match snapshot 1`] = `"<label class=\\"inputs__label \\">undefined<input title=\\"undefined\\" placeholder=\\"undefined\\" class=\\"inputs__input undefined\\" minLength=\\"1\\" maxLength=\\"30\\"/></label>"`;
